packages/components/src/FilterBar/FilterBar.spec.tsx

```diff
@@ -752,4 +754,150 @@ describe("<FilterBar />", () => {
+                >
+                  {item => (
+                    <FilterMultiSelect.Option key={item.key} item={item} />
+                  )}
+                </FilterMultiSelect.ListBoxSection>
+              )}
+            </FilterMultiSelect.ListBox>
+          )}
+        </FilterBar.MultiSelect>
+      )
+    }
+
+    const fetchCityOptions = jest.fn((filterValues: Partial<AsyncValues>) => {
+      const isSupermanInFilterValue = Object.values(filterValues).some(
+        vals => vals.indexOf("3") > -1
+      )
+      const isBatmanInFilterValue = Object.values(filterValues).some(
+        vals => vals.indexOf("4") > -1
+      )
+
+      if (isBatmanInFilterValue && !isSupermanInFilterValue) {
+        return Promise.resolve([{ value: "1", label: "Gotham" }])
+      }
+
+      return Promise.resolve([
+        { value: "1", label: "Gotham" },
+        { value: "2", label: "Metropolis" },
+      ])
+    })
+
+    const fetchHeroOptions = jest.fn((filterValues: Partial<AsyncValues>) => {
+      const isGothamInFilterValue = Object.values(filterValues).some(
+        vals => vals.indexOf("1") > -1
+      )
+      const isMetroInFilterValue = Object.values(filterValues).some(
+        vals => vals.indexOf("2") > -1
+      )
+
+      if (isGothamInFilterValue && !isMetroInFilterValue) {
+        return Promise.resolve([{ value: "4", label: "Batman" }])
+      }
+
+      return Promise.resolve([
+        { value: "3", label: "Superman" },
+        { value: "4", label: "Batman" },
+      ])
+    })
+
+    const config = [
+      {
+        id: "city",
+        name: "City",
+        Component: (
+          <MockFilterAsyncComponent id="city" fetcher={fetchCityOptions} />
+        ),
+      },
+      {
+        id: "hero",
+        name: "Hero",
+        Component: (
+          <MockFilterAsyncComponent id="Hero" fetcher={fetchHeroOptions} />
+        ),
+      },
+    ] satisfies Filters<AsyncValues>
+
+    it("can re-fetch options with all active filter values pulled off of the FilterBarContext", async () => {
+      const { getByRole, queryByRole } = render(
+        <FilterBarWrapper<AsyncValues> filters={config} defaultValues={{}} />
+      )
+
+      // open city filter
+      await user.click(getByRole("button", { name: "City" }))
+
+      // assert both cities are shown
+      await waitFor(() => {
+        expect(queryByRole("option", { name: "Gotham" })).toBeVisible()
+        expect(queryByRole("option", { name: "Metropolis" })).toBeVisible()
+      })
+
+      // select filter option
+      await user.click(getByRole("option", { name: "Gotham" }))
+
+      // close city filter
+      await user.click(document.body)
+
+      // open hero filter
+      await user.click(getByRole("button", { name: "Hero" }))
+
+      // assert only Batman is shown
+      await waitFor(() => {
+        expect(queryByRole("option", { name: "Batman" })).toBeVisible()
+        expect(queryByRole("option", { name: "Superman" })).toBeNull()
```

Comment:
```suggestion
        expect(queryByRole("option", { name: "Superman" })).not.toBeInTheDocument()
```

Reply:
I'm reasonably sure this won't work, since `queryByRole` will return `null` since it's not in the document anywhere. Subsequently, trying to see that `null` is 'not in the document' throws an error.

Reply:
All my other tests where I write it begs to differ 😉 It's probably doing a similar check under the hood, but it's just more readable this way :)